app/src/DataHandler.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import * as d3 from 'd3';
+import landings from './data/landings.csv';
+// FIELDS: Year, State, AFS Name, Pounds, Dollars, TSN, Collection, Confidentiality
+
+class DataHandler extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      year: "all",
+      usState: "all",
+      fish: "all",
+      aggregateBy: "Dollars" // or Pounds or Dollars/Pound
+    }
+
+    this.handleChange = this.handleChange.bind(this)
+  };
+
+  // TODO: how to deal with invalid input?
+  handleChange(event) {
+    const target = event.target
+    const value = target.value
+    const name = target.name
+
+    this.setState({
+      [name]: value
+    })
+  }
+
+  // FILTRATIONS BY STATE
+  // TODO: these can be made generic (i.e. should just be in some componentUpdate function)
+  filterByYear(data) {
+    return this.state.year === "all" ? data : data.filter(d => d['Year'] === this.state.year)
+  };
+
+  filterByState(data) {
+    return this.state.usState === "all" ? data : data.filter(d => d['State'] === this.state.usState)
+  };
+
+  filterByFish(data) {
+    return this.state.fish === "all" ? data : data.filter(d => d['AFS Name'] === this.state.fish)
+  };
+
+  filterNaN = (record) => {
+    return !isNaN(Number(record[1]))
+  }
+
+  // DATA AGGREGATIONS
+  // TODO: some of the state names are messy
```

Comment:
if some state names are messy then you  process landing.csv using python and upload it

Reply:
yeah I'll clean it up, we're expecting them to be all caps like the example in your first comment?